### PR TITLE
Add jobResultObject field

### PR DESCRIPTION
**Pull request by despadam**
## Motivation

#1388 

## Description

A `jobResultObject` field should be included in the `job.schema`, thus persisted in the database. The field contains necessary information for the location of a dataset.

Upon job creation the field should be empty. The value of the field is the result of a dataset archival, meaning we can only have the value after this is successfully executed.

This means that the value can only be provided via the `StatusUpdateJobDto` coming from the archive system.

## Changes:

- Added a non required field `jobResultObject` in the `job.schema`
- Added an optional field `jobResultObject` in the `StatusUpdateJobDto`
- Added the new field in `addStatusFields()` function in `util.ts`
- Passed the extra argument when `addStatusFields()` is called in `jobs.service`. On `create` the field is empty. On `statusUpdate` we pass the value coming from the `StatusUpdateJobDto`.

### CODE REVIEW

File: src/jobs/jobs.service.ts [hunk 1]
```diff
@@ -112,6 +113,7 @@ export class JobsService {
           ),
           statusUpdateJobDto.statusCode,
           statusUpdateJobDto.statusMessage!,
+          statusUpdateJobDto.jobResultObject!,
```
Comment (Junjiequan) [inline src/jobs/jobs.service.ts]: In the StatusUpdateJobDto, both statusMessage and jobResultObject are marked as optional fields, but they are being treated as required fields with the exclamation mark. I’m not sure if I missed something, or if this is a mistake

**Review by Junjiequan — DISMISSED** [review]

File: src/jobs/jobs.service.ts (hunk 1 above)
Comment (despadam) [reply src/jobs/jobs.service.ts]: Thanks! `statusMessage` should be required and `jobResultObject` should be optional. Fixed it the latest commit (https://github.com/SciCatProject/scicat-backend-next/pull/1410/commits/c2af58544f3e8b30fae3664aff7f72a6eed35cf9)

**Review by Junjiequan — APPROVED** [review]
Looks fine to me.

**Review by sbliven — APPROVED** [review]
I tested this out and everything looks good to me!

**Pull request merged**